### PR TITLE
feat(ecma/ast): add `raw` to `NumberLiteral`

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

`raw` to `NumberLiteral`

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_ecma_parser/src/parser/typescript.rs [hunk 1]
```diff
@@ -2046,16 +2046,30 @@ impl<I: Tokens> Parser<I> {
...
                 let lit = self.parse_lit()?;
                 let lit = match lit {
-                    Lit::Num(num) => TsLit::Number(Number {
-                        span: num.span,
-                        value: -num.value,
-                    }),
+                    Lit::Num(Number { span, value, raw }) => {
+                        let mut new_raw = String::from("-");
+                        let orig_raw = match raw {
+                            Some(raw) => raw,
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/parser/typescript.rs:2061]: Please use `push_str` instead.

File: crates/swc_ecma_parser/src/parser/typescript.rs [hunk 2]
```diff
@@ -2046,16 +2046,30 @@ impl<I: Tokens> Parser<I> {
...
+                            _ => value.to_string().into(),
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/parser/typescript.rs:2062]: This line should be a call to `write!` macro.

Comment (kdy1) [reply crates/swc_ecma_parser/src/parser/typescript.rs:2062]: Current code allocates, and acquire global lock in `string_cache`, but we don't need these operations.

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Comment by alexander-akait** [comment]
@kdy1 Done :+1:

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_ast --breaking

**Review by kdy1 — APPROVED** [review]

**Pull request merged**